src/main/java/com/johnstarich/moviematcher/app/MovieMatcherApplication.java

```diff
@@ -53,16 +53,26 @@ public void loginService() {
 		});
 
 		jpost("/login/register", (request, response) -> {
-			if(request.queryParams().size() < 6) return "please fill in all fields";
+			Optional<String> firstname = bodyParam(request, "firstname");
+			Optional<String> lastname = bodyParam(request, "lastname");
+			Optional<String> username = bodyParam(request, "username");
+			Optional<String> password = bodyParam(request, "password");
+			Optional<String> confirmpassword = bodyParam(request, "confirmpassword");
+
+			if(! firstname.isPresent()) throw new HttpException(HttpStatus.BAD_REQUEST, "No firstname provided");
+			if(! lastname.isPresent()) throw new HttpException(HttpStatus.BAD_REQUEST, "No lastname provided");
+			if(! username.isPresent()) throw new HttpException(HttpStatus.BAD_REQUEST, "No username provided");
+			if(! password.isPresent()) throw new HttpException(HttpStatus.BAD_REQUEST, "No password provided");
+			if(! confirmpassword.isPresent()) throw new HttpException(HttpStatus.BAD_REQUEST, "No confirmation provided");
 
-			if(request.queryParams("password").compareTo(request.queryParams("confirmpassword")) == 0) {
+			if(password.get().compareTo(confirmpassword.get()) == 0) {
 				User newUser = new User(
 						new ObjectId(),
-						request.queryParams("username"),
-						request.queryParams("firstname"),
-						request.queryParams("lastname"));
+						username.get(),
+						firstname.get(),
+						lastname.get());
 				try {
-					newUser.register(request.queryParams("password"));
+					newUser.register(password.get());
```

Comment:
Why are we catching this exception? This should probably throw whatever it was trying to throw normally